### PR TITLE
Mark v0.62 as released

### PR DESCRIPTION
**Pull request by arusahni**
### Motivation
  * This PR adds a known-desirable feature.

Update the v0.62 documentation to reflect the release of v0.62.2 to production.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by morsapaes — APPROVED** [review]

File: doc/user/content/releases/v0.62.md [hunk 1]
```diff
@@ -1,7 +1,8 @@
 ---
 title: "Materialize v0.62"
 date: 2023-07-26
-released: false
+released: true
+patch: 2
```
Comment (morsapaes) [inline doc/user/content/releases/v0.62.md:5]:
```suggestion
patch: 3
```

Comment (arusahni) [reply doc/user/content/releases/v0.62.md:5]: @morsapaes while we cut a v0.62.3, it was not released. Does that matter for the release docs?

Comment (morsapaes) [reply doc/user/content/releases/v0.62.md:5]: Sorry, I missed this comment yesterday! The QA team relies on this for CI testing, and AFAIU wants to test against all cut releases, even if these weren't rolled out to users. @philip-stoev can confirm, though!

**Comment by morsapaes** [comment]
Closing this PR since I just put up #20846 with the release notes. Thanks for taking care of it, @arusahni!

**Pull request closed without merging**